travis/build_info.py

```diff
@@ -0,0 +1,207 @@
+    # If org has > 100, this code will need to be modified
+    #  to include 'offset' requests
+    req = requests.get(
+        BASE_URL + 'v3/owner/{org}/repos?active=true'.format(org=org)
+    )
+    if req.status_code != 200:
+        raise requests.HTTPError(req.status_code)
+    req_json = req.json()
+    try:
+        for i in req_json['repositories']:
+            repo_list.append(i['name'])
+    except KeyError:
+        raise KeyError("Cannot parse response")
+
+    return repo_list
+
+
+def get_active_builds(org, repo):
+    """
+    Returns list of active builds for a given repo slug
+    """
+
+    repo_slug = '{org}/{repo}'.format(org=org, repo=repo)
+    req = requests.get(
+        BASE_URL + 'repos/{repo_slug}/builds'.format(repo_slug=repo_slug)
+    )
+    build_list = req.json()
+    active_build_list = []
+    for build in build_list:
+        if build['state'] != 'finished':
+            active_build_list.append(build)
+    return active_build_list
+
+
+# @property
+# def build_id(build_item):
+#
+#     return build_item['id']
+#
+
+def get_jobs(build_id):
+    """
+    Get the jobs for a build
+    return: list of dicts
+    """
+
+    req = requests.get(
+        BASE_URL + 'v3/build/{build_id}/jobs'.format(build_id=build_id)
+    )
+    return req['jobs']
+
+
+# @property
+# def job_state(job):
+#     """
+#     possible states:
+#     * received
+#     * queued
+#     * created
+#     * is there a started?
+#     """
+#
+#     return job['state']
+
+
+def repo_active_build_count(builds):
+    """
+    Returns counts of total builds, and total
+    running builds for a given list of them
+
+    This method assumes it has received a list
+    of active builds.
+    """
+    build_count = 0
+    started_count = 0
+    for build in builds:
+        build_count += 1
+        if build['state'] == 'started':
+            started_count += 1
+
+    return build_count, started_count
+
+
+def repo_active_job_count(active_builds):
+    """
+    Returns count of active jobs associated with builds in a
+    given repo.
+    """
+
+    job_count = 0
+    started_job_count = 0
+    for build in active_builds:
+        if build['stage'] != 'finished':
+            jobs = get_jobs(build['id'])
+            for job in jobs:
+                if job['state'] != 'finished':
+                    job_count += 1
+                    if job['state'] != 'queued':
+                        started_job_count += 1
+
+    return job_count, started_job_count
+
+
+def main(raw_args):
+    """
+    Parse args and execute the script according to those args
+    """
+
+    desc = (
+        "Obtain information on active/waiting Travis builds."
+    )
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument(
+        '--org', '-o',
+        dest='org',
+        help='Travis org',
+        required=True
+        )
+    parser.add_argument(
+        '--task-class',  # this is not doing anything for now.
```

Comment:
Yeah still a to-do. Honestly I'm close to the other pull request and would rather leave this in for now. It is a no-op in this commit anyway.